win32/src/job_object_helper.c

```diff
@@ -11,7 +11,130 @@
+            result = MU_FAILURE;
+        }
+        else
+        {
+            /*Codes_SRS_JOB_OBJECT_HELPER_18_007: [ job_object_helper_limit_resources shall call GetCurrentProcess to get the current process handle. ]*/
+            HANDLE current_process = GetCurrentProcess();
+            if (current_process == NULL)
+            {
+                LogLastError("GetCurrentProcess() failed");
+                /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+                result = MU_FAILURE;
+            }
+            else
+            {
+                /*Codes_SRS_JOB_OBJECT_HELPER_18_008: [ job_object_helper_limit_resources shall call AssignProcessToJobObject to assign the current process to the new job object. ]*/
+                if (AssignProcessToJobObject(job_object, current_process) == FALSE)
+                {
+                    LogLastError("AssignProcessToJobObject(HANDLE job_object = %p, HANDLE current_process = %p) failed", job_object, current_process); 
+                    /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+                    result = MU_FAILURE;
+                }
+                else
+                {
+                    MEMORYSTATUSEX memory_status_ex;
+                    memory_status_ex.dwLength = sizeof(memory_status_ex);
+                    /*Codes_SRS_JOB_OBJECT_HELPER_18_009: [ job_object_helper_limit_resources shall call GlobalMemoryStatusEx to get the total amount of physical memory. ]*/
+                    if (!GlobalMemoryStatusEx(&memory_status_ex))
+                    {
+                        LogLastError("GlobalMemoryStatus(&memory_status_ex = %p) failed", &memory_status_ex);
+                        /*Codes_SRS_JOB_OBJECT_HELPER_18_015: [ If there are any failures, job_object_helper_limit_resources shall fail and return a non-zero value. ]*/
+                        result = MU_FAILURE;
+                    }
+                    else
+                    {
+                        /*Codes_SRS_JOB_OBJECT_HELPER_18_010: [ job_object_helper_limit_resources shall call SetInformationJobObject, passing JobObjectExtendedLimitInformation and a JOBOBJECT_EXTENDED_LIMIT_INFORMATION object with JOB_OBJECT_LIMIT_JOB_MEMORY set and JobMemoryLimit set to the percent_physical_memory percent of the physical memory in bytes. ]*/
+                        JOBOBJECT_EXTENDED_LIMIT_INFORMATION extended_limit_information = {0};
+                        extended_limit_information.BasicLimitInformation.LimitFlags  = JOB_OBJECT_LIMIT_JOB_MEMORY;
+                        extended_limit_information.JobMemoryLimit = percent_physical_memory * memory_status_ex.ullTotalPhys / 100;
```

Comment:
>.JobMemoryLimit = percent_physical_memory [](http://example.com/codeflow?start=50&length=41)

Nice, so, we are looking at physical memory.
This is another nail in the coffin to not have paging enabled. I was pushing for that today.